extensions/db/src/test/java/org/mule/extension/db/integration/AbstractDbIntegrationTestCase.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+package org.mule.extension.db.integration;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.core.IsInstanceOf.instanceOf;
+import static org.junit.Assert.assertThat;
+import static org.mule.extension.db.integration.DbTestUtil.selectData;
+import static org.mule.extension.db.integration.TestRecordUtil.assertRecords;
+import org.mule.extension.db.api.StatementResult;
+import org.mule.extension.db.integration.model.AbstractTestDatabase;
+import org.mule.extension.db.integration.model.Field;
+import org.mule.extension.db.integration.model.Record;
+import org.mule.extension.db.integration.model.derbyutil.DerbyTestStoredProcedure;
+import org.mule.extension.db.internal.domain.connection.DbConnectionProvider;
+import org.mule.functional.junit4.FlowRunner;
+import org.mule.functional.junit4.MuleArtifactFunctionalTestCase;
+import org.mule.functional.junit4.runners.ArtifactClassLoaderRunnerConfig;
+import org.mule.functional.junit4.runners.RunnerDelegateTo;
+import org.mule.runtime.api.message.MuleMessage;
+import org.mule.runtime.extension.api.runtime.ConfigurationProvider;
+
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import javax.sql.DataSource;
+
+import org.junit.Before;
+import org.junit.runners.Parameterized;
+
+@RunnerDelegateTo(Parameterized.class)
+@ArtifactClassLoaderRunnerConfig(exportClasses = {DbConnectionProvider.class, DerbyTestStoredProcedure.class})
+public abstract class AbstractDbIntegrationTestCase extends MuleArtifactFunctionalTestCase {
```

Comment:
We need to have a TestRail project with all the covered scenarios for this connector. Or maybe not for everything but for the things that have changed.